src/main/java/com/venafi/vcert/sdk/connectors/cloud/CloudConnector.java

```diff
@@ -87,25 +104,85 @@ public void authenticate(Authentication auth) throws VCertException {
     this.user = cloud.authorize(auth.apiKey());
   }
 
+  ZoneConfiguration getZoneConfiguration(UserDetails user, CertificatePolicy policy) {
+    ZoneConfiguration zoneConfig = new ZoneConfiguration().customAttributeValues(new HashMap<>());
+    if (policy == null) {
+      return zoneConfig;
+    }
+    zoneConfig.policy(policy.toPolicy());
+    policy.toZoneConfig(zoneConfig);
+    return zoneConfig;
+  }
+
   @Override
-  public ZoneConfiguration readZoneConfiguration(String tag) throws VCertException {
-    VCertException.throwIfNull(tag, "empty zone name");
-    Zone zone = getZoneByTag(tag);
-    CertificatePolicy policy = getPoliciesById(
-        Arrays.asList(zone.defaultCertificateIdentityPolicy(), zone.defaultCertificateUsePolicy()));
-    return zone.getZoneConfiguration(user, policy);
+  public ZoneConfiguration readZoneConfiguration(String zone) throws VCertException {
+    ProjectZone projectZone = null;
+    String[] zoneIdentifiers = parseZoneIdentifiers(zone);
+
+    Projects projects = cloud.projects(auth.apiKey());
+    if (projects.projects().isEmpty()) {
+      throw new VCertException("No projects present.");
+    }
+
+    if (zoneIdentifiers[0] != null) {
+      // Find zone by ID
+      String zoneId = zoneIdentifiers[0];
+
+      for (Project project : projects.projects()) {
+        for (ProjectZone projZone : project.zones()) {
+          if (zoneId.equals(projZone.id())) {
+            projectZone = projZone;
+            break;
+          }
+        }
+      }
```

Comment:
This can be optimised to use the API which finds zone by ID (instead of loading all zones).